internal/rsl/rsl.go

```diff
@@ -513,6 +514,64 @@ func GetFirstReferenceEntryForRef(repo *git.Repository, targetRef string) (*Refe
 	return firstEntry, annotations, nil
 }
 
+// SkipAllInvalidReferenceEntriesForRef identifies invalid RSL reference entries.
+// Each invalid entry points to a target that is not reachable for the current
+// target of the same reference, indicating that history has been rewritten via a
+// rebase for the reference. After the invalid entries are identified, an annotation
+// entry is created that marks all of these entries as to be skipped.
+func SkipAllInvalidReferenceEntriesForRef(repo *git.Repository, targetRef string, signCommit bool) error {
+	slog.Debug("Checking if RSL entries point to non-existent commits...")
+
+	latestEntry, _, err := GetLatestReferenceEntryForRef(repo, targetRef)
+	iterator := Entry(latestEntry)
+	if err != nil {
+		return err
+	}
+
+	entriesToSkip := []plumbing.Hash{}
+
+	for {
+		if entry, ok := iterator.(*ReferenceEntry); ok {
+			latestEntryCommit, err := gitinterface.GetCommit(repo, latestEntry.TargetID)
+
+			var isAncestor bool
+
+			switch {
```

Comment:
I see what you're doing, but I personally find this less readable than regular if else, even if there's a nested if in there. I think there's a more readable sequence to "okay we have an error, is it object not found?" or "we don't have an error, let's check whether there's ancestry" and so on.

Reply:
if you want to keep the switch, even just reordering err == nil and errors.Is(err, obj not found) would go a long way as at least the err != nil cases would be grouped together (with the default).

Reply:
The linter hit me here, telling me to switch to a switch case. If you are ok with it I can add a linter ignore.

Reply:
okay, let's reorder the cases then so err==nil is handled first? also `inAncestor` could be moved up.

Reply:
I have flipped the statements